relay-server/src/actors/events.rs

```diff
@@ -1451,6 +1451,7 @@ impl Handler<HandleEnvelope> for EventManager {
             start_time,
             sampling_project,
         } = message;
+        let event_category = envelope.get_event_category();
```

Comment:
If you look a few lines lower, there is the `is_event` flag with a comment on how this relates to outcomes. You can replace that with your new `event_category` option and keep the comment.

Particularly, you can replace the `if is_event` with the same `if let Some()` or `if event_category.is_some()` that you use in the common endpoint code.